wordpress-mobile/WordPress-Android
# The redesign of comment details content

See https://github.com/orgs/Automattic/projects/944?pane=issue&itemId=60382108
Design: yWt5gg3nWORhu079Qfv3mS-fi-1135_4600
Slack discussion: p1715785205781899/1715757510.947859-slack-C06BWNSR02K

This PR mainly implements the UI redesign of comment details content. I also removed lot of logic related to unused UI components at the moment, some of the logic will come back in modern Kotlin way when I handle the comment moderation tasks.

| Screenshot 1 | Screenshot 2 |
|---|---|
| ![Screenshot_20240516-151102](https://github.com/wordpress-mobile/WordPress-Android/assets/3839951/4348275d-3520-454b-b545-fba51c81d23b) | ![Screenshot_20240516-151120](https://github.com/wordpress-mobile/WordPress-Android/assets/3839951/59636f45-0aa5-470e-a184-fabd6cab8585) |


-----

## To Test:

1. Sign in JP app
2. Go to `Notifications` tab
3. Click on a notification with `Comment` type
4. The UI should be the same as the design on Figma
5. Click on the icon with three dot `...`
6. It should open the comment menu.
7. Click on `User Info`
8. It should display the user info bottom sheet.
9. The functionalities on the bottom sheet should be working as before.
10. Open the comment menu again.
11. Click on `Edit comment`.
12. It should navigate you to the screen of `Edit comment`.
13. The functionalities in this screen should be working as before.
14. Go to My Site -> More -> Comments, click on a comment
15. Repeat step 5 - 13.
16. Done, thank you!

P.S. The `User Info` bottom sheet displays different UI when we are from a comment or a notification because a notification contains richer information in its data.

-----

## Regression Notes

1. Potential unintended areas of impact

    - comment, notification

18. What I did to test those areas of impact (or what existing automated tests I relied on)

    - manual

19. What automated tests I added (or what prevented me from doing so)

    - only updated some broken unit tests

-----

## PR Submission Checklist:

Skipped

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
